src/main/java/com/conveyal/r5/rastercost/CustomCostField.java

```diff
@@ -100,30 +132,51 @@ public int additionalTraversalTimeSeconds (EdgeStore.Edge currentEdge, int baseT
+            customCostValue = 0.0;
+        }
+        // if customCostValue from HashMap was null and flag allowNullcustomCostEdges is false
         if (customCostValue == null) {
             throw new CustomCostFieldException("Custom cost not found for edge with osmId: " + currentEdge.getOSMID());
         }
-        // check the type (Integer or Double) and cast to Double if needed
-        Double customCostFactor;
-        if (customCostValue instanceof Integer) {
-            customCostFactor = ((Integer) customCostValue).doubleValue();
-        } else if (customCostValue instanceof Double) {
-            customCostFactor = (Double) customCostValue;
-        } else {
-            throw new CustomCostFieldException("Unexpected type (should be number) for custom cost with osmId: " + currentEdge.getOSMID());
-        }   
-        // calculate seconds to be added to the base traversal time 
-        // multiply the base travel time with custom cost factor and sensitivity coefficient
-        // this value is then added to the base traversal time
-        Double additionalCostSeconds = baseTraversalTimeSeconds * customCostFactor * this.sensitivityCoefficient;
-        // throw an error if the custom cost is NaN
-        if (Double.isNaN(additionalCostSeconds)) {
-            throw new CustomCostFieldException("Custom cost addition result is NaN for osmId:  " + currentEdge.getOSMID());
+
+        Double additionalCostSeconds;
+        // if customCostValue is 0.0, or was set to 0.0 because was null and flag allowNullCustomCostEdges is true
+        // skip calculating the additional custom cost time and fallback to 0.0
+        if (customCostValue.equals(0.0)) {
+            additionalCostSeconds = 0.0;
+        }
+        // calculate the additionalCostSeconds for edge where customCostValue is found
+        else {
+            // cast to Double if needed
+            Double customCostFactor;
+            if (customCostValue instanceof Integer) {
+                customCostFactor = ((Integer) customCostValue).doubleValue();
+            } else if (customCostValue instanceof Double) {
+                customCostFactor = (Double) customCostValue;
+            } else {
+                throw new CustomCostFieldException("Unexpected type (should be number) for custom cost with osmId: " + currentEdge.getOSMID());
+            }   
+            // calculate seconds to be added to the base traversal time 
+            // multiply the base travel time with custom cost factor and sensitivity coefficient
+            // this value is then added to the base traversal time
+            additionalCostSeconds = baseTraversalTimeSeconds * customCostFactor * this.sensitivityCoefficient;
+            // throw an error if the custom cost is NaN
+            if (Double.isNaN(additionalCostSeconds)) {
+                throw new CustomCostFieldException("Custom cost addition result is NaN for osmId:  " + currentEdge.getOSMID());
```

Comment:
Shouldn’t that already be caught in line 140 - or how do we end up with a NaN here if there has been a non-null value in the custom cost field?